src/WooCommerce.php

```diff
@@ -985,6 +985,41 @@ public static function getTaxonomyTermsAsSelectOptions($taxonomy) {
     return is_wp_error($terms) ? [] : $term_options;
   }
 
+  /**
+   * Retrieves the earliest "back in stock" date from a product or its variations.
+   * If there's stock, the product is considered available and nothing is returned.
+   * @param WC_Product
+   *    The product object.
+   * @return string
+   *    The earliest back in stock date, if any.
+   */
+  public static function getEarliestBackInStock($product) {
+    if ($product->get_stock_quantity() > 0) {
+      return '';
+    }
+
+    $meta_key = '_' . Plugin::PREFIX . '_back_in_stock_date';
+    $back_in_stock_date = get_post_meta($product->get_id(), $meta_key, TRUE) ?? '';
+
+    if ($product->is_type('variable')) {
+      $variations = $product->get_available_variations();
+      foreach ($variations as $variation) {
+        $variation_obj = wc_get_product($variation['variation_id']);
+        $variation_stock_date = get_post_meta($variation['variation_id'], $meta_key, TRUE);
+        // If a variation has stock or doesn't have a "back in stock" date
+        // it's considered to be immediately available for purchase.
+        if ($variation_obj->get_stock_quantity() > 0 || !$variation_stock_date) {
```

Comment:
Do we also need to consider `backorder_allowed`?

Reply:
I don't think we do, since the check was already implemented before calling the `getEarliestBackInStock` function:

https://github.com/netzstrategen/wordpress-shop-standards/pull/142/files#diff-bd92fcb7a0c2355394fe0edb44db3c7d0648b4c4d382c92b18dab73cc4865b35R180

https://github.com/netzstrategen/wordpress-shop-standards/pull/142/files#diff-bd92fcb7a0c2355394fe0edb44db3c7d0648b4c4d382c92b18dab73cc4865b35R180